alphabase/constants/atom.py

```diff
@@ -200,3 +204,77 @@ def calc_mass_from_formula(formula: str):
         mass of the formula
     """
     return np.sum([CHEM_MONO_MASS[elem] * n for elem, n in parse_formula(formula)])
+
+
+class ChemicalCompositonFormula:
+    def __init__(self, formula):
+        self.elements = self._parse_formula(formula)
+
+    @classmethod
+    def from_rdkit_mol(cls, mol: Chem.Mol):
+        if not mol:
+            raise ValueError("Invalid RDKit molecule")
```

Comment:
```suggestion
            raise ValueError(f"Invalid RDKit molecule: {mol}")
```

Reply:
In this case `mol` will be `None`, so we can track the failed molecules only outside of this method
Alternatively, I can modify it to take a SMILES string as the input, so we could have control over meaningful logging

Reply:
`if not ""` would also trigger the log ..

Reply:
Yes, and this is the behaviour that I want to have, I'm just saying that there is no use in using the format string here, unless changing the logic of the argument and going from smiles instead of going from rdkit mol object

Reply:
Rewrote to take smiles as an argument, so we can have more meaningful error messages